src/stores/pacbioRunCreate.js

```diff
@@ -90,53 +133,47 @@ export const usePacbioRunCreateStore = defineStore('pacbioRunCreate', {
   getters: {
     /**
      * Returns a list of all fetched smrt link versions
-     * @param {Object} state The Vuex state object
+     * @param {Object} state the pinia state object
      */
-
     smrtLinkVersionList: (state) => {
       return state.resources.smrtLinkVersions
     },
 
     defaultSmrtLinkVersion: (state) => {
       return Object.values(state.resources.smrtLinkVersions).find((version) => version.default)
     },
+
     // The smrtLinkVersion of the current run
     currentSmrtLinkVersion: (state) => {
       return state.smrtLinkVersion || {}
     },
-    // TODO refactor to reuse the functions
-    // poolsRequest and pools copied over from pools/getters.js
-    poolRequest: () => {
-      const rootStore = useRootStore()
-      return rootStore.api.traction.pacbio.pools
-    },
-    poolsArray: (state) => {
-      return Object.values(state.pools).map((pool) => {
-        const libraries = (pool.libraries || []).map((libraryId) => {
-          const { id, type, request, tag, run_suitability } = state.libraries[libraryId]
-          const { sample_name } = state.requests[request]
-          const { group_id } = state.tags[tag] || {}
-          return { id, type, sample_name, group_id, run_suitability }
-        })
-        const { barcode } = state.tubes[pool.tube]
 
-        return {
-          ...pool,
-          libraries,
-          barcode,
-          run_suitability: {
-            ...pool.run_suitability,
-            formattedErrors: buildRunSuitabilityErrors({ libraries, pool }),
-          },
+    /**
+     * Returns a list of all the tubes with their contents (pool or library)
+     * @param {Object} state the pinia state object
+     * @returns
+     */
+    tubeContents: (state) => {
+      return Object.values(state.tubes).reduce((result, tube) => {
+        // We should assume a tube only has one pool
+        const pool = state.pools[tube.pools?.[0]]
+        const library = state.libraries[tube.libraries]
```

Comment:
Just a doubt- what does 'tube.libraries' return? I hope it is not an array (as for pools)?

Reply:
No its just a singular library. The naming is a bit annoying but its to make it consistent with the service. Libraries is a singular id.